src/main/java/internet/shop/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package internet.shop.dao.impl;
+
+import internet.shop.dao.ManufacturerDao;
+import internet.shop.db.Storage;
+import internet.shop.lib.Dao;
+import internet.shop.model.Manufacturer;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addProduct(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return Storage.getDbManufacturer().stream()
+                .filter(x -> x.getId().equals(id))
+                .findFirst();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.getDbManufacturer();
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        List<Manufacturer> dbManufacturer = Storage.getDbManufacturer();
+        for (int i = 0; i < dbManufacturer.size(); i++) {
+            if (manufacturer.getId().equals(dbManufacturer.get(i).getId())) {
+                dbManufacturer.set(i, manufacturer);
+                return manufacturer;
+            }
+        }
+        return null;
```

Comment:
returning null is a bad practice, better to throw exception

Reply:
done